mlx_vlm/models/hunyuan_vl/vision.py

```diff
@@ -0,0 +1,342 @@
+        self.spatial_merge_size = config.spatial_merge_size
+        self.interpolate_mode = config.interpolate_mode
+
+        self.patch_embedding = nn.Conv2d(
+            in_channels=config.num_channels,
+            out_channels=self.embed_dim,
+            kernel_size=self.patch_size,
+            stride=self.patch_size,
+            bias=True,
+        )
+
+        self.max_num_patches = (config.max_image_size // self.patch_size) ** 2
+        self.num_positions = self.max_num_patches + 1
+        self.position_edge = int(self.num_positions**0.5)
+        self.position_embedding = nn.Embedding(self.num_positions, self.embed_dim)
+
+    def __call__(self, pixel_values: mx.array, grid_thw: list) -> mx.array:
+        num_patches = pixel_values.shape[0]
+        # Reshape: (num_patches, C*P*P) -> (num_patches, C, P, P) -> (num_patches, P, P, C) for MLX conv
+        pixel_values = pixel_values.reshape(
+            num_patches, self.num_channels, self.patch_size, self.patch_size
+        )
+        pixel_values = pixel_values.transpose(0, 2, 3, 1)  # NCHW -> NHWC for MLX
+
+        # Apply patch embedding
+        patch_embeds = self.patch_embedding(pixel_values)  # (N, 1, 1, embed_dim)
+        patch_embeds = patch_embeds.reshape(1, num_patches, self.embed_dim)
+
+        # Get position embeddings and interpolate for each grid
+        pos_embed_weights = self.position_embedding.weight[1:, :]  # Skip cls token
+        base_pos_embed = pos_embed_weights.reshape(
+            1, self.position_edge, self.position_edge, self.embed_dim
+        )
+
+        patch_pos_embed_list = []
+        for grid in grid_thw:
+            t, h, w = grid
+            h_float = float(h) + 0.1
+            w_float = float(w) + 0.1
+
+            # Interpolate position embeddings
+            # Note: MLX doesn't have direct interpolate, so we use a simple approach
+            scale_h = h_float / self.position_edge
+            scale_w = w_float / self.position_edge
+
+            # For simplicity, use nearest neighbor or bilinear interpolation
+            target_h = int(h)
+            target_w = int(w)
+
+            # Simple bilinear interpolation
+            pos_embed = self._interpolate_pos_embed(base_pos_embed, target_h, target_w)
+            pos_embed = pos_embed.reshape(1, -1, self.embed_dim)
+            patch_pos_embed_list.append(pos_embed)
+
+        patch_pos_embed = mx.concatenate(patch_pos_embed_list, axis=1)
+        embeddings = patch_embeds + patch_pos_embed
+
+        return embeddings
+
+    def _interpolate_pos_embed(
+        self, pos_embed: mx.array, target_h: int, target_w: int
+    ) -> mx.array:
+
+        # pos_embed shape: (1, src_h, src_w, embed_dim)
+        src_h = pos_embed.shape[1]
+        src_w = pos_embed.shape[2]
+        embed_dim = pos_embed.shape[3]
+
+        if src_h == target_h and src_w == target_w:
+            return pos_embed
+
+        # Create interpolation grid
+        h_ratio = src_h / target_h
```

Comment:
This needs to be
```
h_ratio = src_h / (target_h + 0.1)
```
and then 
```
src_i = (i + 0.5) * h_ratio - 0.5
```
(same for width)
otherwise quality is broken :-)

Reply:
Thanks, will update and test 👌🏽

Reply:
Could you please send a diff for this?

Reply:
Done

Reply:
Thank you very much!